jbpm-container-test/jbpm-in-container-test/jbpm-container-test-suite/src/test/resources/org/jbpm/test/container/archive/localtransactions/tomcat-context.xml

```diff
@@ -6,10 +6,16 @@
    -->
     <Loader delegate="false"/>
 
-    <Listener className="bitronix.tm.integration.tomcat55.BTMLifecycleListener" />
-
     <Resource name="TransactionSynchronizationRegistry" auth="Container" type="javax.transaction.TransactionSynchronizationRegistry"
-              factory="bitronix.tm.BitronixTransactionSynchronizationRegistryObjectFactory" />
+              factory="org.jboss.narayana.tomcat.jta.TransactionSynchronizationRegistryFactory" />
+
+    <Transaction factory="org.jboss.narayana.tomcat.jta.UserTransactionFactory" />
 
-    <Transaction factory="bitronix.tm.BitronixUserTransactionObjectFactory" />
+    <!-- Database resources -->
+    <Resource name="h2DataSource" uniqueName="h2Datasource" auth="Container" type="org.h2.jdbcx.JdbcDataSource" username="sa"
+              user="sa" password="sa" url="jdbc:h2:mem:test-db;MVCC=TRUE;DB_CLOSE_DELAY=-1" description="H2 Data Source" loginTimeout="0"
+              factory="org.h2.jdbcx.JdbcDataSourceFactory"/>
+    <Resource name="jdbc/jbpm" uniqueName="transactionalDataSource" auth="Container" type="javax.sql.DataSource"
+              username="sa" password="sa" driverClassName="com.arjuna.ats.jdbc.TransactionalDriver"
+              url="jdbc:arjuna:java:comp/env/h2DataSource" description="Transactional Driver Data Source"/>
```

Comment:
I have looked at this and it is still not clear to me how the filtered-resources work sorry :(

I tracked references of the file btm-resources.properties to bitronix.resources.properties in the pom.xml, I can see that is referenced then in btm-config.properties via the property bitronix.tm.resource.configuration. btm-config is referenced back in the pom by the properties bitronix.config.properties which sets a system property bitronix.tm.configuration.

I therefore assume that BTM itself is processing those files so I just not sure what is needed here.

Let me explain the config I have put in the tomcat-context.xml and then you can let me know what you think is best.

Narayana wraps XA datasources within the com.arjuna.ats.jdbc.TransactionalDriver. So in the config I have here I need a standard H2 datasource (h2DataSource), then I need to reference that wrapped by the com.arjuna.ats.jdbc.TransactionalDriver. Arjuna registers for jdbc:arjuna URLS and the rest of the URL is then used to determine the URL of the datasource we need to wrap.

Is this tomcat-context.xml filtered at all I wonder? Perhaps I can use ${org.jbpm.datasource.connection.url} for the H2 URL and ${org.jbpm.datasource.username} and so on directly in that XML file and they might be replaced at build time?